nx_arangodb/classes/graph.py

```diff
@@ -645,12 +699,15 @@ def add_node_override(self, node_for_adding, **attr):
             # attr_dict.update(attr)
 
             # New:
-            self._node[node_for_adding] = self.node_attr_dict_factory()
-            self._node[node_for_adding].update(attr)
+            node_attr_dict = self.node_attr_dict_factory()
+            node_attr_dict.data = attr
+            self._node[node_for_adding] = node_attr_dict
 
             # Reason:
-            # Invoking `update` on the `attr_dict` without `attr_dict.node_id` being set
-            # i.e trying to update a node's attributes before we know _which_ node it is
+            # We can optimize the process of adding a node by creating avoiding
+            # the creation of a new dictionary and updating it with the attributes.
+            # Instead, we can create a new node_attr_dict object and set the attributes
+            # directly. This only makes 1 network call to the database instead of 2.
```

Comment:
I think I've seen this multiple times now. At some point we should try to move duplications into separate methods.

Reply:
Yeah. this specific comment applies to the 4 following methods;

- `nx.Graph.add_node_override`
- `nx.Graph.add_nodes_from_override`
- `nx.DiGraph.add_node_override`
- `nx.DiGraph.add_nodes_from_override`

My intention of duplication here was to keep the "monkey patching" encapsulated per method, instead of trying to shorten code & making the Graph CRUD methods rely on helpers